clients/java/src/main/java/io/unitycatalog/client/auth/TokenProvider.java

```diff
@@ -9,141 +9,135 @@
+   * TokenProvider newProvider = TokenProvider.create(existingProvider.getConfigs());
    * }</pre>
    *
-   * @param configs a map containing authentication configuration parameters
+   * @param configs a map containing authentication configuration parameters. Must contain a
+   *     non-null, non-empty {@code "type"} key.
    * @return a token provider configured according to the provided parameters
-   * @throws IllegalArgumentException if both token and OAuth configs are provided, if OAuth configs
-   *     are incomplete (only some OAuth parameters are set), or if no valid authentication
-   *     configuration is found
-   * @throws NullPointerException if configs is null
+   * @throws IllegalArgumentException if the {@code "type"} parameter is null, empty, or if required
+   *     authentication parameters for the specified type are missing or invalid
+   * @throws RuntimeException if a custom TokenProvider class cannot be instantiated (e.g., class
+   *     not found, no default constructor, or instantiation failure)
    */
-  static TokenProvider createFromConfigs(Map<String, String> configs) {
-    String token = configs.get(AuthProps.TOKEN);
-    String oauthUri = configs.get(AuthProps.OAUTH_URI);
-    String oauthClientId = configs.get(AuthProps.OAUTH_CLIENT_ID);
-    String oauthClientSecret = configs.get(AuthProps.OAUTH_CLIENT_SECRET);
-
-    if (token != null) {
-      Preconditions.checkArgument(
-          oauthUri == null && oauthClientId == null && oauthClientSecret == null,
-          "Invalid Unity Catalog authentication configuration: token-based and OAuth "
-              + "settings were both supplied. Configure exactly one authentication method.");
-      return create(token);
-    }
+  static TokenProvider create(Map<String, String> configs) {
+    String authType = configs.get(AuthConfigs.TYPE);
+    Preconditions.checkArgument(
+        authType != null && !authType.trim().isEmpty(),
+        "Required configuration key '%s' is missing or empty. "
+            + "Must be 'static', 'oauth', or a fully qualified TokenProvider class name.",
+        AuthConfigs.TYPE);
+    switch (authType) {
+      case AuthConfigs.STATIC_TYPE:
+        StaticTokenProvider staticTokenProvider = new StaticTokenProvider();
+        staticTokenProvider.initialize(configs);
+        return staticTokenProvider;
 
-    if (oauthUri != null || oauthClientId != null || oauthClientSecret != null) {
-      Preconditions.checkArgument(
-          oauthUri != null && oauthClientId != null && oauthClientSecret != null,
-          "Incomplete OAuth configuration detected. All of the keys are required: "
-              + "oauthUri, oauthClientId, oauthClientSecret. Please ensure they are "
-              + "all set.");
+      case AuthConfigs.OAUTH_TYPE:
+        OAuthTokenProvider oauthTokenProvider = new OAuthTokenProvider();
+        oauthTokenProvider.initialize(configs);
+        return oauthTokenProvider;
```

Comment:
why not move the initialize to outside this switch case? its common to all cases.